src/Kanbas/Courses/Quizzes/Editor/Details/index.tsx

```diff
@@ -1,11 +1,245 @@
-import { Form } from "react-bootstrap";
+import { Editor } from '@tinymce/tinymce-react';
+import { useEffect, useState } from "react";
+import { Button, Form } from "react-bootstrap";
+import { FaPlus } from 'react-icons/fa';
+import "./index.css"
+import { useNavigate, useParams } from 'react-router-dom';
+import { useDispatch, useSelector } from 'react-redux';
+import { KanbasState } from '../../../../store';
+import { setQuiz, addQuiz, resetQuiz, updateQuiz, fetchQuizzes } from '../../reducer';
+import { eQuizType } from '../../../../store/enums/eQuizType';
+import { eAssignmentGroup } from '../../../../store/enums/eAssignmentGroup';
+
+function QuizDetailsEditor({ isCreate }: { isCreate: boolean }) {
+    // TODO move this later to env file
+    const API_KEY = "mhgsbn0hpoxglrhp0i5yksi6m4rk8nkutj2fn8qxn9cn7s9i";
+
+    const dispatch = useDispatch();
+    const navigate = useNavigate();
+    const { courseId, quizId } = useParams();
+
+    const quizDetails = useSelector(
+        (state: KanbasState) => state.quizzesReducer.quiz
+    );
+
+    const quizTypes = Object.keys(eQuizType).filter(key => isNaN(Number(key)));
+    const assignmentGroup = Object.keys(eAssignmentGroup).filter(key => isNaN(Number(key)));
+
+    const [value, setValue] = useState("");
+    const [text, setText] = useState("");
+
+    const onEditorInputChange = (newValue: any, editor: any) => {
+        setValue(newValue);
+        setText(editor.getContent({ format: "text" }));
+        dispatch(setQuiz({ ...quizDetails, description: newValue }));
+    }
+
+    const formatDate = (date: Date) => {
+        return date ? new Date(date).toISOString().split('T')[0] : '';
+    }
+
+    const handleCancel = () => {
+        navigate(`/Kanbas/Courses/${courseId}/Quizzes`);
+        dispatch(resetQuiz());
+    }
+
+    const handleSave = (isPublish: boolean) => {
+        if (isPublish) {
+            dispatch(setQuiz({ ...quizDetails, isPublish: isPublish }));
+        }
+
+        if (isCreate) {
+            dispatch(addQuiz({ course: courseId }));
+        } else {
+            dispatch(updateQuiz());
+        }
+
+        handleCancel();
+    }
 
-function QuizDetailsEditor() {
     return (
-        <>
-            <Form>
-            </Form>
-        </>
+        <Form>
+            <Form.Group className='mb-3 w-50' controlId="formQuizName">
+                <Form.Control
+                    type="text"
+                    className="form-control"
+                    value={quizDetails.title}
+                    placeholder="Quiz Name"
+                    onChange={(e) => dispatch(setQuiz({
+                        ...quizDetails, title: e.target.value
+                    }))}
+                    onFocus={(e) => e.target.style.textAlign = 'left'}
+                />
+            </Form.Group>
+
+            <Form.Group className='mb-3' controlId='formQuizDescription'>
+                <Editor
```

Comment:
I recommend moving this component usage with all this hardcoding to a /common component and parameterizing it. This way we don't have all the config for this component over here.